app/controllers/pages_controller.rb

```diff
@@ -15,8 +15,12 @@ def index
 
   def room_filters_glossary
     authorize :page
-    @filters = RoomCharacteristic.all.pluck(:chrstc_descr, :chrstc_desc254).uniq.sort
-    characteristics = RoomCharacteristic.all.pluck(:chrstc_descrshort, :chrstc_desc254).uniq.sort
+    @filters = RoomCharacteristic.all.pluck(:chrstc_descr, :chrstc_desc254).uniq
+    @filters.delete_if {|x| x.include?(nil)}
+    @filters.sort
+    characteristics = RoomCharacteristic.all.pluck(:chrstc_descrshort, :chrstc_desc254).uniq
+    characteristics.delete_if {|x| x.include?(nil)}
+    characteristics.sort
```

Comment:
@rsmoke @britaumich - Why do we need to call query on RoomCharacteristic twice ? Can we optimize this code as follows ?

_@filters = RoomCharacteristic.all.pluck(:chrstc_descr, :chrstc_desc254, chrstc_descrshort).uniq
@filters.delete_if {|x| x.include?(nil)}
@filters.sort_

And then on line 25 we can use _@filters.each_